core/src/main/java/io/grpc/util/GracefulSwitchLoadBalancer.java

```diff
@@ -77,21 +76,23 @@ public GracefulSwitchLoadBalancer(Helper helper) {
     this.helper = checkNotNull(helper, "helper");
   }
 
-  /** Gracefully switch to a new load balancing policy. */
-  public void switchTo(LoadBalancerProvider newLbProvider) {
-    checkNotNull(newLbProvider, "newLbProvider");
+  /**
+   * Gracefully switch to a new policy defined by the given factory. Two factories are considered to
```

Comment:
I don't think the second sentence is necessary. It only adds confusion. Other than that, LGTM.

Reply:
Maybe change it to "Gracefully switch to a new policy defined by the given factory, if the given factory isn't equal to the current one".

Reply:
> if the given factory isn't equal to the current one".

hmm, sounds better.